examples/Remarketing/UploadConversionWithIdentifiers.php

```diff
@@ -0,0 +1,252 @@
+<?php
+
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Ads\GoogleAds\Examples\Remarketing;
+
+require __DIR__ . '/../../vendor/autoload.php';
+
+use GetOpt\GetOpt;
+use Google\Ads\GoogleAds\Examples\Utils\ArgumentNames;
+use Google\Ads\GoogleAds\Examples\Utils\ArgumentParser;
+use Google\Ads\GoogleAds\Lib\OAuth2TokenBuilder;
+use Google\Ads\GoogleAds\Lib\V9\GoogleAdsClient;
+use Google\Ads\GoogleAds\Lib\V9\GoogleAdsClientBuilder;
+use Google\Ads\GoogleAds\Lib\V9\GoogleAdsException;
+use Google\Ads\GoogleAds\Util\V9\ResourceNames;
+use Google\Ads\GoogleAds\V9\Common\UserIdentifier;
+use Google\Ads\GoogleAds\V9\Enums\UserIdentifierSourceEnum\UserIdentifierSource;
+use Google\Ads\GoogleAds\V9\Errors\GoogleAdsError;
+use Google\Ads\GoogleAds\V9\Services\ClickConversion;
+use Google\Ads\GoogleAds\V9\Services\ClickConversionResult;
+use Google\Ads\GoogleAds\V9\Services\UploadClickConversionsResponse;
+use Google\ApiCore\ApiException;
+
+/**
+ * Uploads a conversion using hashed email address instead of GCLID.
+ */
+class UploadConversionWithIdentifiers
+{
+    private const CUSTOMER_ID = 'INSERT_CUSTOMER_ID_HERE';
+    private const CONVERSION_ACTION_ID = 'INSERT_CONVERSION_ACTION_ID_HERE';
+    private const EMAIL_ADDRESS = 'INSERT_EMAIL_ADDRESS_HERE';
+    // The date time at which the conversion occurred.
+    // Must be after the click time, and must include the time zone offset.
+    // The format is "yyyy-mm-dd hh:mm:ss+|-hh:mm", e.g. '2019-01-01 12:32:45-08:00'.
+    private const CONVERSION_DATE_TIME = 'INSERT_CONVERSION_DATE_TIME_HERE';
+    private const CONVERSION_VALUE = 'INSERT_CONVERSION_VALUE_HERE';
+
+    // Optional: Specifies the order ID.
+    private const ORDER_ID = 'INSERT_ORDER_ID_HERE';
```

Comment:
should ORDER_ID be null instead?
it's optional

Reply:
+1